### PR TITLE
[chore] Update datadog-agent dependency to the latest commit

### PR DESCRIPTION
**Pull request by dmitryax**
To unblock https://github.com/open-telemetry/opentelemetry-collector/pull/6314

### CODE REVIEW

File: cmd/configschema/go.mod [hunk 1]
```diff
@@ -53,10 +53,10 @@ require (
 	github.com/ClickHouse/clickhouse-go/v2 v2.3.0 // indirect
 	github.com/DataDog/agent-payload/v5 v5.0.34 // indirect
 	github.com/DataDog/datadog-agent/pkg/obfuscate v0.40.0-rc.2.0.20220930142800-72e24cd2e404 // indirect
-	github.com/DataDog/datadog-agent/pkg/otlp/model v0.40.0-rc.2.0.20220930142800-72e24cd2e404 // indirect
-	github.com/DataDog/datadog-agent/pkg/quantile v0.40.0-rc.2.0.20220930142800-72e24cd2e404 // indirect
+	github.com/DataDog/datadog-agent/pkg/otlp/model v0.40.0-rc.2.0.20221013201707-c46e9de16283 // indirect
```
Comment (bogdandrutu) [inline cmd/configschema/go.mod:56]: why not https://github.com/DataDog/datadog-agent/releases/tag/pkg%2Fotlp%2Fmodel%2Fv0.40.0-rc.5?

Comment (dmitryax) [reply cmd/configschema/go.mod:56]: It doesn't include the upgrade to 0.62.0

**Review by mx-psi — APPROVED** [review]
LGTM, I meant to do this yesterday but you were faster than me :)

File: cmd/configschema/go.mod (hunk 1 above)
Comment (mx-psi) [reply cmd/configschema/go.mod:56]: v0.40.0-rc.5 was tagged on the 7.40.x branch. The 0.62.0 bump will not be available on 7.40, but on 7.41. The pseudoversion is a bit confusing...

**Pull request merged**